src/fu-engine.c

```diff
@@ -3319,26 +3350,33 @@ fu_engine_update_history_device (FuEngine *self, FuDevice *dev_history, GError *
 		return TRUE;
```

Comment:
Probably need to explicitly clear the error with `fu_device_set_update_error `here.